src/Services/Utils/UtilsParser.js

```diff
@@ -20,9 +20,29 @@ function currencyConvertParse(json) {
   return json;
 }
 
+function dataTypeParse(json) {
+  json = json['util:ReferenceDataItem'];
+
+  if (Object.prototype.toString.call(json) !== '[object Array]') {
+    throw new UtilsParsingError(json);
+  }
+
+  const props = Object.getOwnPropertyNames(json[0]);
```

Comment:
This does look a bit unsafe. Image first element will have { code, deprecated } only and some of them as you mentioned previously (and [docs as well](https://support.travelport.com/webhelp/uapi/uAPI.htm#../Subsystems/Schemas/Content/Schemas/ReferenceDataRetrieveRsp.html%3FTocPath%3DSchema%7C_____186) ) might have `name` or `description`.

I would probably write it in a way where we handle all cases like

```javascript
const knownFieds = ['Code', 'Deprecated'];
const optionalFields = ['Name', 'Description'];

json = json.map((referenceItem) => {
  const item = {};
  knownFields.forEach(...);

  optionalFields.forEach(field => if (referenceItem[field]) { ... } )

  return item;
   
});
```

Reply:
Yes.. it seems more predicatble.. will do